d2graph/d2graph.go

```diff
@@ -1917,3 +1917,73 @@ func (obj *Object) IterDescendants(apply func(parent, child *Object)) {
 		c.IterDescendants(apply)
 	}
 }
+
+func (obj *Object) ShiftDescendants(dx, dy float64) {
```

Comment:
leave a comment. i specifically want to know from the comment whether it also moves the object itself